### DISCUSSION TITLE
Default slot in RAC

### DISCUSSION THREAD

**Discussion by hosmelq** [adobe/react-spectrum#5142] [Q&A]
Hello guys,

Is there a way to set a default slot within the context? I have two buttons - one with start and end icons and the other with only one icon. I currently use slots to pass props to these icons, but I would like to avoid passing a slot to the button with only an icon.

Examples.

```tsx
<Button>
  <Icon icon={...} slot="start" />
  <Text>Label</Text>
  <Icon icon={...} slot="end" />
</Button>
```

```tsx
<Button>
  <Icon icon={...} />
</Button>
```

I got an error here because a slot prop is required.

**Comment by snowystinger** [comment]
Our Buttons don't have any slots like start or end and we don't have an Icon component in RAC either, can you share a codesandbox with what you're trying to do?

Otherwise, my best guess for this would be you just need to provide a default for the slot prop inside the Icon component

**Reply by hosmelq** [reply]
As you suggested, I added a default slot to the Icon component. Thank You!

**Comment by devongovett** [comment]
There is a `defaultSlot` symbol in our implementation but looks like it's not currently exported from the package: https://github.com/adobe/react-spectrum/blob/d40e4fa6724d2441cee63d5b1f72020a52a57aae/packages/react-aria-components/src/utils.tsx#L27 We could probably export that? We might wanna rename it or something though. We can discuss with the team.

**Reply by hosmelq** [reply]
I need to build the following component, but I have the same issue as the Icon because of the Button since the RAC Calendar only supports the next and previous slots. I can use a native button but would like to reuse the Button component. I think exporting the Symbol would help here, too.

<img width="328" alt="calendar" src="https://github.com/adobe/react-spectrum/assets/1166143/cd6b3071-219c-4c61-9116-81bd979efc87">

**Reply by marbemac** [reply]
Running into the same issue - any chance `defaultSlot` could be exported from react-aria-components?

**Reply by snowystinger** [reply]
See https://github.com/adobe/react-spectrum/issues/5671

**Reply by alexasselin008** [reply]
DEFAULT_SLOT is now exported from react-aria-components as of version 1.2 🎉!
